src/library_fs.js

```diff
@@ -1271,13 +1271,13 @@ mergeInto(LibraryManager.library, {
       FS.mkdev('/dev/tty1', FS.makedev(6, 0));
       // setup /dev/[u]random
       var random_device;
-      if (typeof crypto !== 'undefined') {
+      if (typeof crypto !== 'undefined' && crypto['getRandomValues'] !== 'function') {
         // for modern web browsers
         var randomBuffer = new Uint8Array(1);
         random_device = function() { crypto.getRandomValues(randomBuffer); return randomBuffer[0]; };
       } else if (ENVIRONMENT_IS_NODE) {
-        // for nodejs
-        random_device = function() { return require('crypto')['randomBytes'](1)[0]; };
+        // for nodejs with or without crypto support included
```

Comment:
instead of a try-catch, how about `ENVIRONMENT_IS_NODE && ..check for that..` above?

Reply:
My knowledge is probably not enough but I don't see how to test the `require('crypto')` in the if part.

Reply:
Yeah, good point, that would need a try-catch as well I guess.

Thinking on this some more, it looks like node 4.2.6 has `crypto` in the global namespace already. So I don't think we even need require here? That is, the check a few lines above this can be `ENVIRONMENT_IS_NODE && typeof crypto === 'object'`

Reply:
Well on my Synology NAS (I don't have access to a dev machine right now), crypto is not in the global namespace every time.
```
$ cat foo.js
console.log(typeof crypto)
var c = require('crypto')
console.log(typeof c)
$ node --version
v4.8.4
$ node foo.js
undefined
object
$ node
> typeof crypto
'object'
```
Furthermore in the NodeJs doc they use `try catch` to determine the presence of it: [Determining if crypto support is unavailable](https://nodejs.org/api/crypto.html#crypto_determining_if_crypto_support_is_unavailable)

Reply:
I see, ok.